### DISCUSSION TITLE
Can override/replace IdentifyTenant middleware

### DISCUSSION THREAD

**Discussion by margarizaldi** [filamentphp/filament#7790] [Ideas]
I think some of us need to override the default tenant identification `IdentifyTenant::class` middleware, for example to store tenant in session rather than querying all the time, or for someone who needs more complex implementation.

Something like:
```php
$panel->tenantIdentifierMiddleware(MyMiddleware::class)
```
will be great.

**Comment by pboivin** [comment]
As a workaround, I think you can override the file via composer configuration:
https://downing.tech/posts/overriding-vendor-classes

**Reply by danharrin** [reply]
I would suggest replacing the object using the container instead I think

**Reply by pboivin** [reply]
Much better indeed. Should be possible in this case.

**Reply by margarizaldi** [reply]
> I would suggest replacing the object using the container instead I think

@danharrin I'll be very happy to learn about this one and for now I will try @pboivin solution. Thanks all.

**Reply by margarizaldi** [reply]
> I would suggest replacing the object using the container instead I think

I think I just need to put this line in AppServiceProvider
```php
$this->app->bind(IdentifyTenant::class, CustomIdentifyTenant::class);
```
Hopefully I'm doing the right thing

**Reply by danharrin** [reply]
Yup that is replacing an object in the container

**Reply by margarizaldi** [reply]
Thanks much.
